src/database/src/database.rs

```diff
@@ -0,0 +1,565 @@
+        gate::Gate,
+        gls_navaid::GlsNavaid,
+        ndb_navaid::NdbNavaid,
+        path_point::PathPoint,
+        procedure::{
+            approach::{map_approaches, Approach},
+            arrival::{map_arrivals, Arrival},
+            departure::Departure,
+        },
+        runway::RunwayThreshold,
+        vhf_navaid::VhfNavaid,
+        waypoint::Waypoint,
+    },
+    sql_structs::{self},
+    util,
+};
+
+pub struct Database {
+    database: Option<Connection>,
+}
+
+#[derive(Debug)]
+struct NoDatabaseOpen;
+
+impl Display for NoDatabaseOpen {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result { write!(f, "No database open") }
+}
+
+impl std::error::Error for NoDatabaseOpen {}
+
+impl Database {
+    pub fn new() -> Self { Database { database: None } }
+
+    fn get_database(&self) -> Result<&Connection, NoDatabaseOpen> { self.database.as_ref().ok_or(NoDatabaseOpen) }
+
+    pub fn set_active_database(&mut self, mut path: String) -> Result<(), Box<dyn std::error::Error>> {
+        // Check if the path is a directory and if it is, search for a sqlite file
+        let formatted_path = format!("\\work/{}", path);
+        if util::get_path_type(std::path::Path::new(&formatted_path)) == util::PathType::Directory {
+            path = util::find_sqlite_file(&formatted_path)?;
+        }
+
+        // We have to open with flags because the SQLITE_OPEN_CREATE flag with the default open causes the file to
+        // be overwritten
+        let flags = OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_URI | OpenFlags::SQLITE_OPEN_NO_MUTEX;
+        let conn = Connection::open_with_flags(path, flags)?;
+        self.database = Some(conn);
+
+        Ok(())
+    }
+
+    pub fn execute_sql_query(
+        &self, sql: String, params: Vec<String>,
+    ) -> Result<serde_json::Value, Box<dyn std::error::Error>> {
+        // Execute query
+        let conn = self.get_database()?;
+        let mut stmt = conn.prepare(&sql)?;
+        let names = stmt
+            .column_names()
+            .into_iter()
+            .map(|n| n.to_string())
+            .collect::<Vec<_>>();
+
+        // Collect data to be returned
+        let data_iter = stmt.query_map(params_from_iter(params), |row| {
+            let mut map = serde_json::Map::new();
+            for (i, name) in names.iter().enumerate() {
+                let value = match row.get_ref(i)? {
+                    ValueRef::Text(text) => Some(serde_json::Value::String(String::from_utf8(text.into()).unwrap())),
+                    ValueRef::Integer(int) => Some(serde_json::Value::Number(serde_json::Number::from(int))),
+                    ValueRef::Real(real) => {
+                        Some(serde_json::Value::Number(serde_json::Number::from_f64(real).unwrap()))
+                    },
+                    ValueRef::Null => None,
+                    ValueRef::Blob(_) => panic!("Unexpected value type Blob"),
+                };

```

Comment:
Maybe we should import `serde_json::Value` and `serde_json::Number` into scope to remove the repetition?